### PR TITLE
fix: report syntax error

### PR DESCRIPTION
**Pull request by hyoban**
### Description

When writing some CSS demos in markdown files, the syntax may not always be valid. Then we can not lint anymore.

```
Oops! Something went wrong! :(

ESLint: 9.2.0

SyntaxError: CssSyntaxError: Unknown word (29:5)
  27 |     }
  28 |
> 29 |     ...
     |     ^
  30 |   }
  31 |   // #endregion
  32 |
    at Xf (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/plugins/postcss.mjs:49:10786)
    at Ks (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/plugins/postcss.mjs:49:17948)
    at Object.ol [as parse] (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/plugins/postcss.mjs:49:18271)
    at parse4 (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/index.mjs:20716:24)
    at async coreFormat (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/index.mjs:21206:7)
    at async formatWithCursor (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/index.mjs:21408:14)
    at async Module.format2 (file:///Users/hyoban/i/BewlyBewly/node_modules/.pnpm/prettier@3.3.0/node_modules/prettier/index.mjs:22793:25)
```

### CODE REVIEW

**Comment by antfu** [comment]
I would usually put `<!-- eslint-skip -->` above the block - as you won't expect any linting anyway for invalid code. Or you can have `/* ... */` instead. 

I am not sure if silently bail out is a good idea or not. But maybe a try...catch is still useful that we can emit an ESLint error to say `Failed to parse the code` instead of crashing

**Comment by hyoban** [comment]
Updated to report parse error

File: src/rules/prettier.ts [hunk 1]
```diff
@@ -36,12 +36,23 @@ export default {
...
+            ...(context.options[0] || {}),
+          })
 
-        reportDifferences(context, sourceCode, formatted)
+          reportDifferences(context, sourceCode, formatted)
+        }
+        catch (error) {
+          context.report({
+            loc: {
+              start: { line: 1, column: 0 },
```
Comment (antfu) [inline src/rules/prettier.ts]: Is it possible to get the position information from Prettier or the error project and report it to the place? And then we could fallback to the start of the file if failed

Comment (hyoban) [reply src/rules/prettier.ts]: The error message looks like this:

```
CssSyntaxError: Unknown word (3:3)
  1 | .foo {
  2 |   color: green;
> 3 |   ...
    |   ^
  4 | }
  5 |
```

I've updated it to get a more accurate location.

File: src/rules/prettier.ts [hunk 2]
```diff
@@ -36,12 +36,36 @@ export default {
...
+                start: { line: +line, column: +column },
+                end: { line: +line, column: +column },
+              }
+            }
+          }
+
+          context.report({
+            loc,
+            message: 'Failed to format the code',
+          })
```
Comment (XiNiHa) [inline src/rules/prettier.ts]: I'd recommend you to look into this: [eslint-plugin-prettier handling errors thrown by Prettier](https://github.com/prettier/eslint-plugin-prettier/blob/de9751c85d059678904035322501bfce120f61b7/eslint-plugin-prettier.js#L209-L233)
(Actually I was going to submit a PR with these changes and just found this PR :joy:)

Comment (hyoban) [reply src/rules/prettier.ts]: Updated to use the same logic with `eslint-plugin-prettier`. Thank you!

**Review by antfu — APPROVED** [review]

**Pull request merged**